plaid/src/apis/npm/npm_web_client.rs

```diff
@@ -0,0 +1,833 @@
+            .ok_or(ApiError::NpmError(NpmError::FailedToDeletePackage))?
+            .as_str()
+            .to_string();
+
+        // Step 2. Perform the actual package deletion with a POST request to /delete
+        let scoped_package_name = format!("@{}/{}", self.config.npm_scope, package);
+        let csrf_token = self
+            .get_csrftoken_from_cookies()
+            .map_err(|_| ApiError::NpmError(NpmError::WrongClientStatus))?;
+        let form_data: HashMap<&str, &str> = [
+            ("package", scoped_package_name.as_str()),
+            ("dsrManifestHash", &dsr_manifest_hash),
+            ("csrftoken", &csrf_token),
+        ]
+        .into();
+        self.client
+            .post(format!(
+                "{}/package/%40{}%2F{}/delete",
+                NPMJS_COM_URL, self.config.npm_scope, package
+            ))
+            .form(&form_data)
+            .send()
+            .await
+            .map(|_| Ok(0))
+            .map_err(|_| ApiError::NpmError(NpmError::FailedToDeletePackage))?
+    }
+
+    /// Add a user to an npm team
+    pub async fn add_user_to_team(&self, params: &str, module: &str) -> Result<i32, ApiError> {
+        self.login()
+            .await
+            .map_err(|_| ApiError::NpmError(NpmError::LoginFlowError))?;
+        let params: AddRemoveUserToFromTeamParams =
+            serde_json::from_str(params).map_err(|_| ApiError::BadRequest)?;
+        let user = self.validate_npm_username(&params.user)?;
+        let team = self.validate_npm_team_name(&params.team)?;
+
+        info!(
+            "Adding user [{}] to team [{}] on behalf of [{module}]",
+            user, team
+        );
+
+        let csrf_token = self
+            .get_csrftoken_from_cookies()
+            .map_err(|_| ApiError::NpmError(NpmError::WrongClientStatus))?;
+        let body = format!(r#"{{"csrftoken": "{}", "user": "{}"}}"#, &csrf_token, user);
+        self.client
+            .post(format!(
+                "{}/settings/{}/teams/team/{}/users",
+                NPMJS_COM_URL, self.config.npm_scope, team
+            ))
+            .header("Content-Type", "text/plain;charset=UTF-8")
+            .body(body)
+            .send()
+            .await
+            .map(|_| Ok(0))
+            .map_err(|_| ApiError::NpmError(NpmError::FailedToAddUserToTeam))?
+    }
+
+    /// Remove a user from an npm team
+    pub async fn remove_user_from_team(&self, params: &str, module: &str) -> Result<i32, ApiError> {
+        self.login()
+            .await
+            .map_err(|_| ApiError::NpmError(NpmError::LoginFlowError))?;
+        let params: AddRemoveUserToFromTeamParams =
+            serde_json::from_str(params).map_err(|_| ApiError::BadRequest)?;
+        let user = self.validate_npm_username(&params.user)?;
+        let team = self.validate_npm_team_name(&params.team)?;
+
+        info!("Removing [{user}] from [{team}] on behalf of [{module}]");
+
+        let body = format!(
```

Comment:
Again, worried about this kind of interpolation.

Reply:
Fixed, plus another one in the previous function